### PR TITLE
[Validator] fix IBAN validator fails if IBAN contains non-breaking space

### LINKED ISSUE #57364: [Validator] IBAN validator fails if IBAN contains non-breaking space
### Symfony version(s) affected

6.4.8

### Description

IBAN validator fails if IBAN is copy/pasted from a source that formatted the IBAN with non-breaking space as separator. This is common on web banking apps and pdf documents issued by banks.

For instance, `BE00 0000 0000 0000`, with hex string `42453030c2a030303030c2a030303030c2a030303030`, fails to validate (assuming the IBAN is otherwise valid which is not the case with this sample IBAN :slightly_smiling_face: ).

The `Iban` validator only strips plain ASCII spaces.

### How to reproduce

Generate a random but correct looking iban such as `"FR841450800040\xc2\xa08782897835E04"` with at least one non-breaking space (\xc2\xa0) somewhere in the string.

Validate that string with the `Iban` validator. Notice the IBAN is reported as invalid.

### Possible Solution

Add non-breaking space to the stripped characters in https://github.com/symfony/symfony/blob/678abb4b128c0bd9f6db83a280bd5355ce234aec/src/Symfony/Component/Validator/Constraints/IbanValidator.php#L183

This change fixed the issue for me:

```php
$canonicalized = str_replace([' ', "\xc2\xa0"], [''], strtoupper($value));
```

It might not be the most elegant solution though :slightly_smiling_face: 

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by antten**
| Q             | A
| ------------- | ---
| Branch?       | 5.4
| Bug fix?      | yes
| New feature?  | no
| Deprecations? | no
| Issues        | Fix #57364
| License       | MIT

Hello,
It's my first contribution to Symfony and I am really happy to be able to participate in the project.

After analyzing the class \Symfony\Component\Validator\Constraints\IbanValidator, I think the solution proposed in the issue is the right one.

### CODE REVIEW

File: src/Symfony/Component/Validator/Constraints/IbanValidator.php [hunk 1]
```diff
@@ -163,9 +163,6 @@ class IbanValidator extends ConstraintValidator
         'YT' => 'FR\d{2}\d{5}\d{5}[\dA-Z]{11}\d{2}', // France
     ];
 
-    /**
-     * {@inheritdoc}
-     */
```
Comment (fabpot) [inline src/Symfony/Component/Validator/Constraints/IbanValidator.php]: I suppose that this comes from fabbot. This should be reverted as not related to this PR.

File: src/Symfony/Component/Validator/Constraints/IbanValidator.php [hunk 2]
```diff
@@ -182,8 +179,8 @@ public function validate($value, Constraint $constraint)
 
         $value = (string) $value;
 
-        // Remove spaces and convert to uppercase
-        $canonicalized = str_replace(' ', '', strtoupper($value));
+        // Remove spaces and non-breaking spaces and convert to uppercase
+        $canonicalized = str_replace([' ', '\XC2\XA0'], '', strtoupper($value));
```
Comment (fabpot) [inline src/Symfony/Component/Validator/Constraints/IbanValidator.php]:
```suggestion
        // Remove spaces (regular, non-breaking, and narrow non-breaking) and convert to uppercase
        $canonicalized = str_replace([' ', "\xc2\xa0", "\xe2\x80\xaf"], '', strtoupper($value));
```

**Review by fabpot — CHANGES_REQUESTED** [review]

File: src/Symfony/Component/Validator/Constraints/IbanValidator.php (hunk 2 above)
Comment (antten) [reply src/Symfony/Component/Validator/Constraints/IbanValidator.php]: Thanks for the suggestion.

**Review by xabbuh — APPROVED** [review]
LGTM, I am just not sure if this qualifies as a bugfix after all

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @antten.

**Comment by xabbuh** [comment]
changelog entry added in 62fcf934ae201622f618fa08358823aa5c828ec3

**Comment by nicodemuz** [comment]
@fabpot should the documentation be updated to warn that the validator allows for various types of whitespaces? Now there is a situation in my project where IBAN numbers stored in the database have non-breaking spaces, etc. This is causing issues when integrating further with other APIs.

Maybe worth to mention in the documentation that the validator accepts various types of human readable IBAN formats.

**Comment by OskarStark** [comment]
PR welcome 👍

**Comment by nicodemuz** [comment]
> PR welcome 👍

@OskarStark See https://github.com/symfony/symfony-docs/pull/21049

**Pull request merged**